src/components/overview/OverviewMain.jsx

```diff
@@ -202,6 +197,19 @@ class overviewMain extends React.Component {
       });
   }
 
+  componentDidUpdate(prevProps) {
```

Comment:
Glad to see everyone found an efficient way to conditionally re-render when we switch product ids in the router.